utils/helm/speckle-server/templates/_helpers.tpl

```diff
@@ -93,3 +93,137 @@ Part-of label
 {{- define "speckle.labels.part-of" -}}
 app.kubernetes.io/part-of: {{ include "speckle.name" . }}
 {{- end }}
+
+{{/*
+Connects to kube api-server to determine if Cilium CRD are present.
+If they are we assume that Cilium is installed.
+*/}}
+{{- define "speckle.networkpolicy.ciliumIsPresent" -}}
+
+{{- end }}
+
+{{/*
+Creates a network policy egress definition for connecting to Redis
+
+Expects the global context "$" to be passed as the parameter
+*/}}
+{{- define "speckle.networkpolicy.egress.redis" -}}
+{{- $port := (default "6379" .Values.redis.networkPolicy.port ) -}}
+{{- if .Values.redis.networkPolicy.inCluster.enabled -}}
+{{ include "speckle.networkpolicy.egress.internal" (dict "podSelector" .Values.redis.networkPolicy.inCluster.podSelector "namespaceSelector" .Values.redis.networkPolicy.inCluster.namespaceSelector "port" $port) }}
+{{- else if .Values.redis.networkPolicy.externalToCluster.enabled -}}
+{{ include "speckle.networkpolicy.egress.external" (dict "ip" .Values.redis.networkPolicy.externalToCluster.ipv4 "port" $port) }}
+{{- end -}}
+{{- end }}
+
+{{/*
+Creates a network policy egress definition for connecting to Postgres
+*/}}
+{{- define "speckle.networkpolicy.egress.postgres" -}}
+{{- $port := (default "5432" .Values.db.networkPolicy.port ) -}}
+{{- if .Values.db.networkPolicy.inCluster.enabled -}}
+{{ include "speckle.networkpolicy.egress.internal" (dict "podSelector" .Values.db.networkPolicy.inCluster.podSelector "namespaceSelector" .Values.db.networkPolicy.inCluster.namespaceSelector "port" $port) }}
+{{- else if .Values.db.networkPolicy.externalToCluster.enabled -}}
+{{ include "speckle.networkpolicy.egress.external" (dict "ip" .Values.db.networkPolicy.externalToCluster.ipv4 "port" $port) }}
+{{- end -}}
+{{- end }}
+
+{{/*
+Creates a network policy egress definition for connecting to Postgres
+*/}}
+{{- define "speckle.networkpolicy.egress.blob_storage" -}}
+{{- $port := (default "443" .Values.s3.networkPolicy.port ) -}}
+{{- if .Values.s3.networkPolicy.inCluster.enabled -}}
+{{ include "speckle.networkpolicy.egress.internal" (dict "podSelector" .Values.s3.networkPolicy.inCluster.podSelector "namespaceSelector" .Values.s3.networkPolicy.inCluster.namespaceSelector "port" $port) }}
+{{- else if .Values.s3.networkPolicy.externalToCluster.enabled -}}
+  {{- $host := (urlParse .Values.s3.endpoint).host -}}
+  {{- $ip := "" -}}
+  {{- if eq (include "speckle.isIPv4" $host) "true" -}}
```

Comment:
Struggled to have the include return a boolean (i.e. not a string) and avoid using `eq` against a string value of "true".  Kept failing for some reason I was unable to debug.